channeld/channeld.c

```diff
@@ -543,6 +589,10 @@ static void channel_announcement_negotiate(struct peer *peer)
 	if (!peer->channel_ready[LOCAL] || !peer->channel_ready[REMOTE])
 		return;
 
+	/* Don't announce channel if we're in stfu mode */
+	if (peer->stfu_request || is_stfu_active(peer))
```

Comment:
In the current code I'm setting `stfu_request` (being renamed to `want_stfu`) to false when STFU is successful.

Maybe we can skip that and leave it on permanently. I don't remember the exact situation I was trying to solve by setting it to false on success (probably trying to make error messages more clear).

I will get rid of turning it to false so we can simplify this if statement and see if it breaks things.

Reply:
Investigating this further: `on_stfu_success` and `stfu_wait_single_msg` both require detecting the entrance into `stfu` mode.

Since `maybe_send_stfu` is called quite frequently, these events will trigger sporadically without some way of separating "waiting for stfu to start" and "stfu is active" modes.